src/main/java/io/github/mirromutth/r2dbc/mysql/ServerVersion.java

```diff
@@ -65,25 +107,19 @@ public static ServerVersion parse(ByteBuf version) {
      * @throws IllegalArgumentException if any version part is negative integer.
      */
     public static ServerVersion create(int major, int minor, int patch) {
-        require(major >= 0, "major version must not be a negative integer");
-        require(minor >= 0, "minor version must not be a negative integer");
-        require(patch >= 0, "patch version must not be a negative integer");
-
-        if (major == 0 && minor == 0 && patch == 0) {
-            return NONE;
-        }
-
-        return new ServerVersion(major, minor, patch);
+        return create0(major, minor, patch, "");
     }
 
     public int compareTo(int major, int minor, int patch) {
         if (this.major != major) {
             return (this.major < major) ? -1 : 1;
         } else if (this.minor != minor) {
             return (this.minor < minor) ? -1 : 1;
+        } else if (this.patch != patch) {
```

Comment:
How about introducing helper methods like `isGreater`/`isLessThan` (see https://github.com/spring-projects/spring-data-commons/blob/master/src/main/java/org/springframework/data/util/Version.java for example). Using `compareTo` is quite error prone and having more expressive methods is beneficial, even though it's internal API.